src/transformers/models/llama/modeling_llama.py

```diff
@@ -300,6 +300,7 @@ def forward(
         hidden_states: torch.Tensor,
         attention_mask: Optional[torch.Tensor] = None,
         position_ids: Optional[torch.LongTensor] = None,
+        cache_position: Optional[torch.LongTensor] = None,
```

Comment:
Can we derive `cache_position` inside the attn layer, instead of passing it as an argument? It would be better for BC purposes :) 

Or is this [passing `cache_position` around] a performance-related change?

Reply:
it's more like we can't create a tensor in the attention, it will be ignore by the graph and will always use the same in and out

Reply:
The cache_positons are only used if you have past key values / a causal mask in sdpa. More over it can be safely None! As only used on the static cache, dynamic will not care

Reply:
Slicing of the causal mask in sdpa and eager should indeed make sure that they don't slice if cache positions is None! Good catch there